libraries/redshop/helper/discount.php

```diff
@@ -110,7 +110,10 @@ public static function getDiscountPriceBaseDiscountDate($productId)
         $today = time();
 
         // Convert discount_enddate to middle night
-        $productData->discount_enddate = RedshopHelperDatetime::generateTimestamp($productData->discount_enddate);
+        if ($productData->discount_endate != '0')
```

Comment:
@NguyenBao10 : please also check `isset($productData->discount_endate)` then to `!= '0'` to avoid notice & warning from PHP in case it's not exist (not set)